src/js/utils/browser.js

```diff
@@ -5,196 +5,233 @@
-  const result = (/MSIE\s(\d+)\.\d/).exec(USER_AGENT);
-  let version = result && parseFloat(result[1]);
-
-  if (!version && (/Trident\/7.0/i).test(USER_AGENT) && (/rv:11.0/).test(USER_AGENT)) {
-    // IE 11 has a different user agent string than other IE versions
-    version = 11.0;
-  }
-
-  return version;
-}());
+export let IE_VERSION = null;
 
 /**
  * Whether or not this is desktop Safari.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_SAFARI = (/Safari/i).test(USER_AGENT) && !IS_CHROME && !IS_ANDROID && !IS_EDGE;
+export let IS_SAFARI = false;
 
 /**
  * Whether or not this is a Windows machine.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_WINDOWS = (/Windows/i).test(USER_AGENT);
+export let IS_WINDOWS = false;
 
 /**
- * Whether or not this device is touch-enabled.
+ * Whether or not this device is an iPad.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const TOUCH_ENABLED = Boolean(Dom.isReal() && (
-  'ontouchstart' in window ||
-  window.navigator.maxTouchPoints ||
-  window.DocumentTouch && window.document instanceof window.DocumentTouch));
+export let IS_IPAD = false;
 
 /**
- * Whether or not this device is an iPad.
+ * Whether or not this device is an iPhone.
  *
  * @static
- * @const
  * @type {Boolean}
  */
-export const IS_IPAD = (/iPad/i).test(USER_AGENT) ||
-  (IS_SAFARI && TOUCH_ENABLED && !(/iPhone/i).test(USER_AGENT));
+// The Facebook app's UIWebView identifies as both an iPhone and iPad, so
+// to identify iPhones, we need to exclude iPads.
+// http://artsy.github.io/blog/2012/10/18/the-perils-of-ios-user-agent-sniffing/
+export let IS_IPHONE = false;
 
 /**
- * Whether or not this device is an iPhone.
+ * Whether or not this device is touch-enabled.
  *
  * @static
  * @const
  * @type {Boolean}
  */
-// The Facebook app's UIWebView identifies as both an iPhone and iPad, so
-// to identify iPhones, we need to exclude iPads.
-// http://artsy.github.io/blog/2012/10/18/the-perils-of-ios-user-agent-sniffing/
-export const IS_IPHONE = (/iPhone/i).test(USER_AGENT) && !IS_IPAD;
+export const TOUCH_ENABLED = Boolean(Dom.isReal() && (
+  'ontouchstart' in window ||
+  window.navigator.maxTouchPoints ||
+  window.DocumentTouch && window.document instanceof window.DocumentTouch));
+
+const UAD = window.navigator && window.navigator.userAgentData;
+
+if (UAD) {
+  // If userAgentData is present, use it instead of userAgent to avoid warnings
+  // Currently only implemented on Chromium
+  // userAgentData does not expose Android version, so ANDROID_VERSION remains `null`
+
+  IS_ANDROID = UAD.platform === 'Android';
+  IS_EDGE = Boolean(UAD.brands.find(b => b.brand === 'Microsoft Edge'));
+  IS_CHROMIUM = Boolean(UAD.brands.find(b => b.brand === 'Chromium'));
+  IS_CHROME = !IS_EDGE && IS_CHROMIUM;
+  CHROME_VERSION = (UAD.brands.find(b => b.brand === 'Chromium') || {}).version || null;
+  IS_WINDOWS = UAD.platform === 'Windows';
+}
+
+if (!CHROME_VERSION) {
```

Comment:
I'm a bit confused about why this block is so long and how `!CHROME_VERSION` acts as a proxy for all the things inside here. Should it be `CHROMIUM_VERSION` instead?

Reply:
Only Chromium browsers support `userAgentData` now. The tests in this block are repeating the same tests with userAgent (for Chromium < 90) or for things that can't be true if `userAgentData` is set and it's Chromium. We can use `IS_CHROMIUM` here as it's a bit clearer.

If and when other browsers add `userAgentData` and we know the values e.g. if Firefox's brand string would be "Mozilla Firefox" or "Firefox" then this will need to be updated.